pipeline_tools/confirm_submission.py

```diff
@@ -2,67 +2,81 @@
 
 import requests
 import argparse
-from .dcp_utils import check_status
-from tenacity import retry, retry_if_result, retry_if_exception_type, wait_exponential, stop_after_delay, RetryError
+from tenacity import retry, retry_if_result, retry_if_exception_type, RetryError, before
+from pipeline_tools.http_requests import HttpRequests
 
-RETRY_SECONDS = 10
-TIMEOUT_SECONDS = 600
 
-
-def status_is_invalid(status):
-    return status != 'Valid'
-
-
-@retry(reraise=True, wait=wait_exponential(multiplier=1, max=RETRY_SECONDS), stop=stop_after_delay(TIMEOUT_SECONDS),
-       retry=(retry_if_result(status_is_invalid) | retry_if_exception_type(requests.HTTPError)))
-def wait_for_valid_status(envelope_url):
+def wait_for_valid_status(envelope_url, http_requests):
     """
     Check the status of the submission. Retry until the status is "Valid", or if there is an error with the request to
     get the submission envelope.
 
-    :param str envelope_url: Submission envelope url
-    :return: str status: Status of the submission ("Valid", "Validating", etc.)
-    """
-    print('Getting status for {}'.format(envelope_url))
-    envelope_js = get_envelope_json(envelope_url)
-    status = envelope_js.get('submissionState')
-    print('submissionState: {}'.format(status))
-    return status
+    Args:
+        envelope_url: Submission envelope url
+        http_requests: HttpRequests object
 
+    Returns:
+        Status of the submission ("Valid", "Validating", etc.)
+
+    Raises:
+        requests.HTTPError if response status indicates an error
+    """
+    def log_before(envelope_url):
+        print('Getting status for {}'.format(envelope_url))
```

Comment:
I've seen this `log_before` function elsewhere in this PR, how about taking it out of this nested functions and making it more general? like: 
```python
    def log_before(current_stage, envelope_url):
        print('Getting'{0}' for {1}'.format(current_stage, envelope_url))
```

Reply:
It's a good thought but I'm not sure that will generalize well beyond these two cases and the code saved is very minimal. I started writing a more generalizable function like
```python
def log_before(format_str, *args):
    print(format_str.format(*args))
```

but didn't go through with it because although it reduces duplication it makes things more verbose in some ways, so it didn't seem like a win overall.

Reply:
That makes sense!